tests/backends/postgresql/test_operations.py

```diff
@@ -78,3 +79,49 @@ def test_prepare_join_on_clause_different_types(self):
         self.assertEqual(
             rhs_expr, Cast(Col(book_table, book_fk_field), author_id_field)
         )
+
+    def test_bulk_batch_size(self):
+        objects = range(2**16)
+        max_query_params = 2**16 - 1
+        first_name_field = Person._meta.get_field("first_name")
+        last_name_field = Person._meta.get_field("last_name")
+        composite_pk = models.CompositePrimaryKey("first_name", "last_name")
+        composite_pk.fields = [first_name_field, last_name_field]
+
+        self.assertEqual(connection.ops.bulk_batch_size([], objects), len(objects))
+
+        # Without server-side binding.
+        with mock.patch.object(
+            type(connection.features),
+            "max_query_params",
+            new_callable=mock.PropertyMock,
+            return_value=None,
+        ):
```

Comment:
Then we could just mock `uses_server_side_binding`, which I think would improve the test a little bit.